docs/requests.md

```diff
@@ -116,6 +116,45 @@ In some cases such as long-polling, or streaming responses you might need to
 determine if the client has dropped the connection. You can determine this
 state with `disconnected = await request.is_disconnected()`.
 
+#### Request Files
+
+Request files are normally sent as multipart form data (`multipart/form-data`).
+
+When you call `await request.form()` you receive a `starlette.datastructures.FormData` which is an immutable
+multidict, containing both file uploads and text input.
+
+When one of the fields is a file upload, you will get a `starlette.datastructures.UploadFile`.
+
+`UploadFile` has the following attributes:
+
+* `filename`: A `str` with the original file name that was uploaded (e.g. `myimage.jpg`).
+* `content_type`: A `str` with the content type (MIME type / media type) (e.g. `image/jpeg`).
+* `file`: A <a href="https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile" target="_blank">`SpooledTemporaryFile`</a> (a <a href="https://docs.python.org/3/glossary.html#term-file-like-object" target="_blank">file-like</a> object). This is the actual Python file that you can pass directly to other functions or libraries that expect a "file-like" object.
+
+
+`UploadFile` has the following `async` methods. They all call the corresponding file methods underneath (using the internal `SpooledTemporaryFile`) in a threadpool.
```

Comment:
I don't think we need to mention the threadpool - at this level of docs we can consider that as an implementation detail.

Reply:
Sure. Updated it. Let me know if you prefer to delete the details about the <a href="https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile" target="_blank">`SpooledTemporaryFile`</a>.